common/oatbox/cache/GcpTokenCacheItemPool.php

```diff
@@ -0,0 +1,251 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA
+ */
+
+declare(strict_types=1);
+
+namespace oat\oatbox\cache;
+
+use oat\oatbox\cache\factory\CacheItemPoolFactory;
+use oat\oatbox\service\ConfigurableService;
+use Psr\Cache\CacheItemInterface;
+use Psr\Cache\CacheItemPoolInterface;
+
+class GcpTokenCacheItemPool extends ConfigurableService implements CacheItemPoolInterface
+{
+    public const SERVICE_ID = 'generis/GcpTokenCacheItemPool';
+
+    public const OPTION_PERSISTENCE = 'persistence';
+    public const OPTION_ENABLE_DEBUG = 'enableDebug';
```

Comment:
Today we already have a constant `DEBUG_MODE` for this purpose. I agree that it is not the ideal, but we should not duplicate it. We should have a centralized place to switch the platform on debug or not

Reply:
Hmm. I will not use this global constant here. I could use at the `conf.php`, but still is a global constant what we want to avoid. This config is based on DI though. This is not code duplication.

Reply:
I understand and I would not recommend to use the global constant. But if someone is changing to true or false, it will not have any effect on your service.

Reply:
I added in the conf.php file, but if you do not mind will not add this to the class cause we are making an effort to remove global constants from our codebase.

Reply:
Instead of constant `DEBUG_MODE`,  `\oat\tao\model\service\ApplicationService::isDebugMode` may be used

Reply:
Closing, cause I removed!